prusti-interface/src/utils.rs

```diff
@@ -297,28 +297,43 @@ pub fn has_extern_spec_attr(attrs: &[ast::Attribute]) -> bool {
     has_prusti_attr(attrs, "extern_spec")
 }
 
-// pub fn get_attr_value(attr: &ast::Attribute) -> String {
-//     use syntax::parse::token;
-//     use syntax::tokenstream::TokenTree;
-
-//     let trees: Vec<_> = attr.tokens.trees().collect();
-//     assert_eq!(trees.len(), 2);
-
-//     match trees[0] {
-//         TokenTree::Token(_, ref token) => assert_eq!(*token, token::Token::Eq),
-//         _ => unreachable!(),
-//     };
+/// Read the value stored in a Prusti attribute (e.g. `prusti::<attr_name>="...")`.
+pub fn read_prusti_attrs(attr_name: &str, attrs: &[ast::Attribute]) -> Vec<String> {
+    let mut strings = vec![];
+    for attr in attrs {
+        if let ast::AttrKind::Normal(ast::AttrItem {
+                                         path: ast::Path { span: _, segments, tokens: _ },
+                                         args: ast::MacArgs::Eq(_, tokens),
+                                         tokens: _,
+                                     }) = &attr.kind {
+            // Skip attributes whose path don't match with "prusti::<attr_name>"
+            if !(
+                segments.len() == 2 &&
+                    segments[0].ident.name.with(|name| name == "prusti") &&
+                    segments[1].ident.name.with(|name| name == attr_name)
+            ) {
+                continue;
+            }
+            use rustc_ast::token::Lit;
+            use rustc_ast::token::Token;
+            use rustc_ast::token::TokenKind;
+            use rustc_ast::tokenstream::TokenTree;
+            match &tokens.0[0].0 {
+                TokenTree::Token(Token {
+                                     kind: TokenKind::Literal(Lit { symbol, .. }),
+                                     ..
+                                 }) => {
+                    strings.push(symbol.as_str().replace("\\\"", "\""))
```

Comment:
This seems a bit fragile?

Reply:
I agree, but I don't have a quick solution. In this PR I just changed the return type from `Option<_>` to `Vec<_>`, moving the function out of `prusti-interface/src/specs/mod.rs`.